File: JavaScript/0014_longest_common_prefix/solution.js
function find_longest_prefix(words) {
    let len_dict = new Object();
    for (let word of words) {
        len_dict[word.length] = word;
    }
    let refer = len_dict[Math.min(...Object.keys(len_dict))];
    let result = new String();
    for (let index = 0; index < refer.length; index ++) {
        let check_set = new Set();
        for (let word of words) {
            check_set.add(word[index]);
        }
        if (check_set.size !== 1) {
            return result;
        }
        let curr = Array.from(check_set);
        result += curr[0];
    }
    return result;
}

const words  = ["flower","flow","flight"];
console.log(find_longest_prefix(words));

File: Python/0014_longest_common_prefix/solution.py
def find_longest_prefix(words):
    len_dict = dict()
    for word in words:
        len_dict[len(word)] = word
    refer = len_dict[min(len_dict)]
    result = str()
    for index in range(len(refer)):
        check_set = set()
        for word in words:
            check_set.add(word[index])
        if len(check_set) != 1:
            return result
        current = list(check_set)
        result += current[0]
    return result

words  = ["flower","flow","flight"]
print(find_longest_prefix(words))